### ISSUE TITLE
.put wrongly treats a String value in a .get chain as Object -> silent loss of data

### ISSUE THREAD

**Issue by leonardpauli** [amark/gun#711]
**TL;DR: `.put` wrongly treats a String value in a .get chain as an Object, thus silently failing to save value.**

```
gun = new Gun()
gun.get('root').get('a').put('text')
gun.get('root').get('a').once(console.log) // "text" a // correct

gun.get('root').get('a').get('b').put('foo')
gun.get('root').get('a').once(console.log) // "text" a // still?
gun.get('root').get('a').get('b').once(console.log) // undefined b // whaa? not saved?? expected "foo" b

gun._.graph // well, it has created a node with {b: "foo"}, but root.a is still "text"

gun.get('root').get('a').put({b: 'bar'}) // putting without an "intermediate" chain
gun.get('root').get('a').once(console.log) // {b: "bar"} a // changed!
gun.get('root').get('a').get('b').once(console.log) // "bar" b // so this works!

gun._.graph // the previous faild-to-link {b: "foo"} is still there, together with the new successful {b: "bar"} (and {root: {a: {#: bSoul}}})
```

Two alternative solutions:
- quick: only allow object as value in .put
- better: make sure non-object values don't prevent intermediate chain nodes creation when using put

<h5>I looked through the code a bit, looks like it checks if there is a value at all instead of checking if the value is of object/partial or Node/gun type; (though was a bit hard with all the cryptic variable names, lack of comments and coherent formatting. That's another issue, though pretty sure it would allow more potential contributors / turn off less of them)</h5>
